File: app/models/film.rb
# frozen_string_literal: true

class Film < ApplicationRecord
  validates :title, :episode_id, :opening_crawl, :director, :producer, :release_date, presence: true
  validates :episode_id, numericality: { only_integer: true }

  has_many :characters_films
  has_many :characters, through: :characters_films

  has_many :films_species
  has_many :species, through: :films_species

  has_many :films_planets
  has_many :planets, through: :films_planets
end

File: app/javascript/components/Film.js
import React from "react"
import {Link} from 'react-router-dom'

function Film(props) {
  return (
    <tr>
      <td><Link to={`/Films/${props.film.id}`}>{props.film.title}</Link></td>
      <td>{props.film.episode_id}</td>
      <td>{props.film.director}</td>
      <td>{props.film.producer}</td>
    </tr>
  );
}

export default Film
